build_system.sh

```diff
@@ -38,38 +52,54 @@ if [ ! -f $UBUNTU_FILE ]; then
   curl -C - -o $UBUNTU_FILE $UBUNTU_BASE_URL/$UBUNTU_FILE --silent
 fi
 
-if [ "$ARCH" != "arm64" ] && [ "$ARCH" != "aarch64" ]; then
-  # Register qemu multiarch
-  docker run --rm --privileged multiarch/qemu-user-static:register --reset
+# Register qemu multiarch
+if [ "$ARCH" = "x86_64" ]; then
+  echo "Registering qemu-user-static"
+  docker run --rm --privileged multiarch/qemu-user-static:register --reset > /dev/null
 fi
 
-# Start docker build
-echo "Building image"
-export DOCKER_CLI_EXPERIMENTAL=enabled
+# Start agnos-builder docker build and create container
+echo "Building agnos-builder docker image"
 docker build -f Dockerfile.agnos -t agnos-builder $DIR
+echo "Creating agnos-builder container"
+CONTAINER_ID=$(docker container create --entrypoint /bin/bash agnos-builder:latest)
 
-# Setup mount container
-MOUNT_CONTAINER_ID=$(docker run -d --privileged --volume $BUILD_DIR:$BUILD_DIR ubuntu:latest sleep infinity)
+# Setup mount container for macOS and CI support (namespace.so)
+if ! docker inspect agnos-mount &>/dev/null; then
+  echo "Building agnos-mount docker image"
+  docker build -f Dockerfile.sparsify -t agnos-mount $DIR
+fi
+echo "Starting agnos-mount container"
+MOUNT_CONTAINER_ID=$(docker run -d --privileged -v $DIR:$DIR agnos-mount)
+
+# Cleanup containers on possible exit
+trap "echo \"Cleaning up containers:\"; \
+docker container rm -f $CONTAINER_ID $MOUNT_CONTAINER_ID" EXIT
+
+# Create host user in container (fixes namespace.so error)
+USERNAME=$(whoami)
+docker exec $MOUNT_CONTAINER_ID bash -c "useradd --uid $(id -u) -U -m $USERNAME" &> /dev/null
 
 # Create filesystem ext4 image
 echo "Creating empty filesystem"
-docker exec $MOUNT_CONTAINER_ID fallocate -l $ROOTFS_IMAGE_SIZE $ROOTFS_IMAGE
-docker exec $MOUNT_CONTAINER_ID mkfs.ext4 $ROOTFS_IMAGE > /dev/null
+docker exec -u $USERNAME $MOUNT_CONTAINER_ID fallocate -l $ROOTFS_IMAGE_SIZE $ROOTFS_IMAGE
```

Comment:
i'd pull `docker exec -u $USERNAME $MOUNT_CONTAINER_ID` into a helper. it's super unreadable as is